connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectorConfig.java

```diff
@@ -68,24 +78,142 @@
+                        if (transformAliases.size() > new HashSet<>(transformAliases).size()) {
+                            throw new ConfigException(name, value, "Duplicate alias provided.");
+                        }
+                    }
+                }, Importance.LOW, TRANSFORMS_DOC, TRANSFORMS_GROUP, 6, Width.LONG, TRANSFORMS_DISPLAY);
     }
 
     public ConnectorConfig() {
         this(new HashMap<String, String>());
     }
 
     public ConnectorConfig(Map<String, String> props) {
-        super(configDef(), props);
+        this(configDef(), props);
+    }
+
+    public ConnectorConfig(ConfigDef configDef, Map<String, String> props) {
+        super(enrich(configDef, props, true), props);
     }
 
-    public ConnectorConfig(ConfigDef subClassConfig, Map<String, String> props) {
-        super(subClassConfig, props);
+    /**
+     * Returns the initialized list of {@link Transformation} which are specified in {@link #TRANSFORMS_CONFIG}.
+     */
+    public <R extends ConnectRecord<R>> List<Transformation<R>> transformations() {
+        final List<String> transformAliases = getList(TRANSFORMS_CONFIG);
+        if (transformAliases == null || transformAliases.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        final List<Transformation<R>> transformations = new ArrayList<>(transformAliases.size());
+        for (String alias : transformAliases) {
+            final String prefix = TRANSFORMS_CONFIG + "." + alias + ".";
+            final Transformation<R> transformation;
+            try {
+                transformation = getClass(prefix + "type").asSubclass(Transformation.class).newInstance();
+            } catch (Exception e) {
+                throw new ConnectException(e);
+            }
+            transformation.configure(originalsWithPrefix(prefix));
+            transformations.add(transformation);
+        }
+
+        return transformations;
     }
-}
+
+    /**
+     * Returns an enriched {@link ConfigDef} building upon the {@code ConfigDef}, using the current configuration specified in {@code props} as an input.
+     * <p>
+     * {@code requireFullConfig} specifies whether required config values that are missing should cause an exception to be thrown.
+     */
+    public static ConfigDef enrich(ConfigDef baseConfigDef, Map<String, String> props, boolean requireFullConfig) {
+        final List<String> transformAliases = (List<String>) ConfigDef.parseType(TRANSFORMS_CONFIG, props.get(TRANSFORMS_CONFIG), Type.LIST);
+        if (transformAliases == null || transformAliases.isEmpty()) {
+            return baseConfigDef;
+        }
+
+        final ConfigDef newDef = new ConfigDef(baseConfigDef);
+
+        for (String alias : new LinkedHashSet<>(transformAliases)) {
+            final String prefix = TRANSFORMS_CONFIG + "." + alias + ".";
+            final String groupPrefix = TRANSFORMS_GROUP + ": " + alias;
```

Comment:
As an arg to `newDef.embed()`, it can be a prefix (if a transformation's `ConfigDef` key spec has group name, otherwise it acts as the group). But it makes sense to just call it `group` here so will do.